src/aleph/netty.clj

```diff
@@ -1053,18 +1082,51 @@
 
 (def ^:no-doc ^String client-event-thread-pool-name "aleph-netty-client-event-pool")
 
-(def ^:no-doc epoll-client-group
+(def ^:private epoll-client-group
   (delay
     (let [thread-count (get-default-event-loop-threads)
           thread-factory (enumerating-thread-factory client-event-thread-pool-name true)]
       (EpollEventLoopGroup. (long thread-count) thread-factory))))
 
-(def ^:no-doc nio-client-group
+(def ^:private nio-client-group
   (delay
     (let [thread-count (get-default-event-loop-threads)
           thread-factory (enumerating-thread-factory client-event-thread-pool-name true)]
       (NioEventLoopGroup. (long thread-count) thread-factory))))
 
+(def ^:private kqueue-client-group
+  (delay
+    (let [thread-count (get-default-event-loop-threads)
+          thread-factory (enumerating-thread-factory client-event-thread-pool-name true)]
+      (KQueueEventLoopGroup. (long thread-count) thread-factory))))
+
+(def ^:private io-uring-client-group
+  (delay
+    (let [thread-count (get-default-event-loop-threads)
+          thread-factory (enumerating-thread-factory client-event-thread-pool-name true)]
+      (IOUringEventLoopGroup. (long thread-count) thread-factory))))
+
+(defn ^:no-doc transport-client-group [transport]
+  (case transport
+    :epoll    epoll-client-group
+    :kqueue   kqueue-client-group
+    :io-uring io-uring-client-group
+    :nio      nio-client-group))
+
+(defn ^:no-doc transport-event-loop-group [transport ^long num-threads ^ThreadFactory thread-factory]
+  (case transport
+    :epoll    (EpollEventLoopGroup. num-threads thread-factory)
+    :kqueue   (KQueueEventLoopGroup. num-threads thread-factory)
+    :io-uring (IOUringEventLoopGroup. num-threads thread-factory)
+    :nio      (NioEventLoopGroup. num-threads thread-factory)))
+
+(defn ^:no-doc transport-server-channel [transport]
+  (case transport
+    :epoll    EpollServerSocketChannel
+    :kqueue   KQueueServerSocketChannel
+    :io-uring IOUringServerSocketChannel
+    :nio      NioServerSocketChannel))
+
```

Comment:
It feels like we have a bunch of these maps, maybe there should be a ns-wide transport map that groups the values together?

Reply:
That approach minimizes the number of changes across all the namespaces.
I don't think we'll have to support another transport, even so, the changes will be pretty obvious.
I would advice keeping it like this.

Anyway, do not hesitate to push commits on top of this PR with your approach. If I do it myself, I would expect a lot of back and forth.

Reply:
This works, too.